og_ui/src/Plugin/Field/FieldFormatter/GroupSubscribeFormatter.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og_ui\Plugin\Field\FieldFormatter\GroupSubscribeFormatter.
+ */
+
+namespace Drupal\og_ui\Plugin\Field\FieldFormatter;
+
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\og\Og;
+use Drupal\user\EntityOwnerInterface;
+
+/**
+ * Plugin implementation for the OG subscribe formatter.
+ *
+ * @FieldFormatter(
+ *   id = "og_ui_group_subscribe",
+ *   label = @Translation("OG Group subscribe"),
+ *   description = @Translation("Display OG Group subscribe links."),
+ *   field_types = {
+ *     "og_group"
+ *   }
+ * )
+ */
+class GroupSubscribeFormatter extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    $entity = $items->getEntity();
+    // @todo Inject this properly.
```

Comment:
Yes.  This is still a WIP :)
On 27 Nov 2015 20:38, "Amitai Burstein" <notifications@github.com> wrote:

> In og_ui/src/Plugin/Field/FieldFormatter/GroupSubscribeFormatter.php
> <https://github.com/amitaibu/og/pull/40#discussion_r46069381>:
>
> > + *   description = @Translation("Display OG Group subscribe links."),
> > + *   field_types = {
> > + *     "og_group"
> > + *   }
> > + * )
> > + */
> > +class GroupSubscribeFormatter extends FormatterBase {
> > +
> > +  /**
> > +   * {@inheritdoc}
> > +   */
> > +  public function viewElements(FieldItemListInterface $items, $langcode) {
> > +    $elements = [];
> > +
> > +    $entity = $items->getEntity();
> > +    // @todo Inject this properly.
>
> possible to do this in this PR?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amitaibu/og/pull/40/files#r46069381>.
>